include/seqan3/io/sequence/sequence_file_out.hpp

```diff
@@ -841,32 +861,57 @@ class sequence_file_out
+                       detail::decays_to_ignore_v<reference_t<quals_t>>),
+                  "You may not select field::SEQ_QUAL and either of field::SEQ and field::QUAL at the same time.");
+
         // make sure that std::ignore inputs are actually range of std::ignore
         assert(!format.valueless_by_exception());
         std::visit([&] (sequence_file_out_format_concept & f)
         {
-            auto seqs_it       = ranges::begin(seqs);
-            auto ids_it        = ranges::begin(ids);
-            auto quals_it      = ranges::begin(quals);
-            auto seq_quals_it  = ranges::begin(seq_quals);
+            if constexpr (!detail::decays_to_ignore_v<reference_t<seq_quals_t>>)
+            {
+                auto ids_it        = ranges::begin(ids);
+                auto seq_quals_it  = ranges::begin(seq_quals);
 
-            auto seqs_end      = ranges::end(seqs);
-            auto ids_end       = ranges::end(ids);
-            auto quals_end     = ranges::end(quals);
-            auto seq_quals_end = ranges::end(seq_quals);
+                auto ids_end       = ranges::end(ids);
+                auto seq_quals_end = ranges::end(seq_quals);
 
-            do
+                do
+                {
+                    f.write(stream,
+                            options,
+                            *seq_quals_it | view::convert<typename reference_t<seq_quals_t>::sequence_alphabet_type>, //TODO view::get<0> instead?
+                            *ids_it,
+                            *seq_quals_it | view::convert<typename reference_t<seq_quals_t>::quality_alphabet_type>);
+
+                    ++ids_it;
+                    ++seq_quals_it;
+                } while ((ids_it != ids_end)     &&
+                         (seq_quals_it != seq_quals_end));
+            }
+            else
             {
-                f.write(stream, options, *seqs_it, *ids_it, *quals_it, *seq_quals_it);
-
-                ++seqs_it;
-                ++ids_it;
-                ++quals_it;
-                ++seq_quals_it;
-            } while ((seqs_it != seqs_end)   &&
-                     (ids_it != ids_end)     &&
-                     (quals_it != quals_end) &&
-                     (seq_quals_it != seq_quals_end));
+                auto seqs_it       = ranges::begin(seqs);
+                auto ids_it        = ranges::begin(ids);
+                auto quals_it      = ranges::begin(quals);
+
+                auto seqs_end      = ranges::end(seqs);
+                auto ids_end       = ranges::end(ids);
+                auto quals_end     = ranges::end(quals);
+
+                do
+                {
+                    f.write(stream, options, *seqs_it, *ids_it, *quals_it);
+
+                    ++seqs_it;
+                    ++ids_it;
+                    ++quals_it;
+                } while ((seqs_it != seqs_end)   &&
+                         (ids_it != ids_end)     &&
+                         (quals_it != quals_end));
+            }
```

Comment:
You could use zip_iterator here, couldn't you?

Reply:
yes, good catch :+1: